src/transformStateWithClones.js

```diff
@@ -7,7 +7,48 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateActions = [];
+  const stateCopy = {
+    ...state,
+  };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        addProperties(stateCopy, action.extraData);
+        break;
+
+      case 'removeProperties':
+        removeProperties(stateCopy, action.keysToRemove);
+        break;
+
+      case 'clear':
+        clearProperties(stateCopy);
+        break;
+
+      default:
+        return `Error, unknown action type: ${action.type}`;
```

Comment:
[CODE STYLE]: The `default` case in a `switch` statement should not return an error message directly. It's better to throw an error or handle it in a way that doesn't interrupt the execution flow of the program. Consider logging the error and continuing with a default action or throwing an exception if the unknown action type is truly exceptional.